surface/CMakeLists.txt

```
@@ -45,11 +47,11 @@ if(build)
             src/vtk_smoothing/vtk_mesh_smoothing_windowed_sinc.cpp)
         
         if("${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}" VERSION_LESS "6.0")
-            set(VTK_SMOOTHING_TARGET_LINK_LIBRARIES vtkCommon vtkWidgets vtkGraphics)
+            set(VTK_SMOOTHING_TARGET_LINK_LIBRARIES pcl_io vtkCommon vtkWidgets vtkGraphics)
         else("${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}" VERSION_LESS "6.0")
-            set(VTK_SMOOTHING_TARGET_LINK_LIBRARIES vtkCommonCore vtkCommonDataModel vtkCommonExecutionModel vtkFiltersModeling)
+            set(VTK_SMOOTHING_TARGET_LINK_LIBRARIES pcl_io vtkCommonCore vtkCommonDataModel vtkCommonExecutionModel vtkFiltersModeling)
         endif("${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}" VERSION_LESS "6.0")
-    endif()
+    endif(IO_STATUS AND VTK_FOUND AND NOT ANDROID)
```

One final detail from my side. We're no longer applying this pattern

```cmake
if(CONDITION)
else(CONDITION)
endif(CONDITION)
```
This was required in older cmake versions this practice was dropped.
```suggestion
    endif()
```